registry/handlers/manifests.go

```diff
@@ -481,15 +482,31 @@ func (imh *manifestHandler) applyResourcePolicy(manifest distribution.Manifest)
 
 }
 
-// DeleteManifest removes the manifest with the given digest from the registry.
+// DeleteManifest removes the manifest with the given digest or the tag with the given name from the registry.
 func (imh *manifestHandler) DeleteManifest(w http.ResponseWriter, r *http.Request) {
 	dcontext.GetLogger(imh).Debug("DeleteImageManifest")
 
-	if imh.Tag != "" {
+	if imh.App.isCache {
 		imh.Errors = append(imh.Errors, errcode.ErrorCodeUnsupported)
 		return
 	}
 
+	if imh.Tag != "" {
+		tagService := imh.Repository.Tags(imh.Context)
+		if err := tagService.Untag(imh.Context, imh.Tag); err != nil {
+			switch err.(type) {
+			case distribution.ErrTagUnknown:
```

Comment:
The spec does not explicitly say what should be returned when the Tag is not found, but I think leaving this case like this (returning without updating errors or setting `StatusAccepted` header) is not very useful.
We should either group this with the `driver.PathNotFoundError` case (correct semantically), or maybe define a specific error for this case (not sure if this makes much sense, though) or even chuck `fallthrough` here.

Reply:
> The spec does not explicitly say what should be returned when the Tag is not found

It's not mentioned in [Deleting tags](https://github.com/opencontainers/distribution-spec/blob/v1.0.0/spec.md#deleting-tags), but looking at [Endpoints](https://github.com/opencontainers/distribution-spec/blob/v1.0.0/spec.md#endpoints), we do see the `404` for this route. So, given the lack of a more specific behavior for tags, we should follow the endpoint spec and keep it consistent.

> maybe define a specific error for this case

The OCI spec has no specific error for this, and the closest not found error defined for this endpoint is `MANIFEST_UNKNOWN`, so although odd, we should keep it IMO.

> I think leaving this case like this (returning without updating errors or setting StatusAccepted header) is not very useful

Oops, I didn't realize this. Good catch. Fixed 👍